torch/csrc/distributed/c10d/NCCLUtils.hpp

```diff
@@ -924,6 +1004,44 @@ struct NCCLTraceBuffer {
     return pg_config;
   }
 
+  const std::map<std::string, std::map<std::string, std::string>>
```

Comment:
Should this logic be in a cpp file instead of a header?

Reply:
Moved all of NCCLTraceBuffer struct methods into .cpp

Reply:
These methods are part of `NCCLTraceBuffer` struct. I will move these into the .cpp file in a separate diff to make this diff easier to review.